Prevent crash if negative change in sweep

AdamISZ opened the pull request:
Fixes #976. Joinmarket will allow, but warn, when
non-zero change values (which are not included in
the transaction) result from sweeps (due to the
impossibility of knowing the exact value in advance).
However, prior to this commit, if that estimation
inaccuracy resulted in a negative change value that
was to be ignored, there was a crash due to the fact
that `jmbitcoin.amount_to_str` does not allow negative
values. Hence we instead print the number in the warning
without going through this formatting function.

AdamISZ commented:
Reproducible error described here: https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/976#issuecomment-895118959 is fixed by this.

undeath commented:
Since you already mentioned how you reproduced this in a test case, how about adding a test case for this?

Fix looks okay to me. The code is already handling out-of-range change amounts well (positive and negative), so really only the logger is causing a problem here.

AdamISZ commented:
> Since you already mentioned how you reproduced this in a test case, how about adding a test case for this?

A test case of what though? If I had changed the behaviour of `jmbitcoin.amount_to_str` to allow negative values, then it would indeed warrant a test case. But I just removed that function call, here.

undeath commented:
A test case to verify that `receive_utxos()` does correctly handle non-zero change amounts on sweeps.

AdamISZ commented:
> A test case to verify that `receive_utxos()` does correctly handle non-zero change amounts on sweeps.

I see, you mean, as well as the existing non-zero change sweep value case, I could add one which is negative. Yes, could be done, currently there is no different functionality between positive and negative.

undeath commented:
Yes, one with a negative value would be nice to prevent this problem from popping up again.

AdamISZ commented:
A test case added. It is as dumb if not more so than the existing test cases :) But after spending some time on thinking about better ways I just went for that copy/paste option.

undeath commented:
Thanks! Looks good to me.

kristapsk reviewed (approved):
I would have fixed `amount_to_str()` instead, but not using it for this single message seems ok to me too, utACK 4586e34ef53d0ad4af017066c69b7afce20ac5e8.
